src/fermilib/utils/_parallel_bubble_sort.py

```diff
@@ -0,0 +1,114 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+"""This module has functions to perform parallel bubble sort on arrays."""
+from functools import partial
+from itertools import product
+
+
+def index(coordinate_ordering, system_side_length, position):
+    """Return the index of an n-dimensional position in an 1-dimensional
+    array.
+
+    The index is computed using the given ordering of coordinate
+    weights as well as the side length of the system from which the
+    n-dimensional position is taken.
+
+    Args:
+        position: The vector (list) of integers indicating the position
```

Comment:
We are trying (not always succeeding) to stick with the convention of describing args by putting their type in parenthesis before the colon, e.g.
position (list): The vector of integers...